Overhaul TLS and ingress certificates docs

mamachanko opened the pull request:
* Adds an inventory of components, how they use ingress+TLS
* Mentions wildcard certificates and their second-class support
* Improves structure and presentation
* Adds more specific examples of ingress issuer configurations
* Refers from the hardening guide

---
### Target

Targeted for TAP `1.5`

### Which other branches do you want a technical writer to cherry-pick this PR to (if any)?

None

<!--
It's best to PR to the most recent relevant branch and leave all the cherry-picking to the
writer, except where earlier branches require factual changes to the content.
For more information about the branches, see https://github.com/pivotal/docs-tap#branches
-->

Petahhh commented on security-and-compliance/tls-and-certificates/ingress/issuer.hbs.md:
```diff
@@ -0,0 +1,268 @@
+
+1. Apply TAP's installation values
+
+Once the configuration is applied, components eventually obtain certificates
```
Can you briefly describe how one would conveniently verify the certificates have been updated? It'll be helpful  but I don't think it's absolutely necessary. I find it personally helpful at the end of a procedure to have some hints if I completed it correctly

Petahhh reviewed (approved):
Thank you for consolidating all this information! It was clear and easy to follow.

One thing I worry about (and this is not particular to you but maybe a system doc maintenance challenge) is how we ensure that big table is accurately maintained. This is not a blocking ask nor do I need a response but something I thought I'd highlight.

mamachanko commented:
Thank you @Petahhh !

> One thing I worry about (and this is not particular to you but maybe a system doc maintenance challenge) is how we ensure that big table is accurately maintained. 

You are right in calling out the risk of documentation drift. My current strategy is: it's maintained by me / my team and we create awareness by sharing edits with the respective teams.

kcoriordan commented:
Hi @mamachanko 
here is the link to the updated table, let me know what you think: https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Application-Platform/1.6/tap/security-and-compliance-tls-and-certificates-ingress-inventory.html

mamachanko commented on security-and-compliance/tls-and-certificates/ingress/inventory.hbs.md:
```diff
@@ -0,0 +1,64 @@
+Package name | Profiles | Has ingress | Ingress purpose | Supports ingress issuer | Supports wildcards | # of ingress | SANs | TLS Documentation
+---|---|---|---|---|---|---|---|---|
+accelerator.apps.tanzu.vmware.com | full, light, view | | | | | | | |
+api-portal.tanzu.vmware.com | full, view | ✅ | Serves the API portal | |  | `1` | `api-portal.<ingress-domain>` (configurable) | [docs](../../../api-portal/about.hbs.md)
```
Fair point. They are, one way or another, all configurable, `sso.apps.tanzu.vmware.com` included. The components generally break up into two camps:
* Those which install a single ingress resource in the form of `<component>.<domain_name>`, e.g. TAP GUI
* Those which install an ingress resource per API instance which gets templated from a `domain_template` feeding `<domain_name>`, e.g. `cnrs.tanzu.vmware.com` and `sso.apps.tanzu.vmware.com`

kcoriordan replied:
I added this as a note at the bottom of the table._

mamachanko commented on security-and-compliance/tls-and-certificates/ingress/inventory.hbs.md:
```diff
@@ -0,0 +1,64 @@
+Package name | Profiles | Has ingress | Ingress purpose | Supports ingress issuer | Supports wildcards | # of ingress | SANs | TLS Documentation
+---|---|---|---|---|---|---|---|---|
+accelerator.apps.tanzu.vmware.com | full, light, view | | | | | | | |
+api-portal.tanzu.vmware.com | full, view | ✅ | Serves the API portal | |  | `1` | `api-portal.<ingress-domain>` (configurable) | [docs](../../../api-portal/about.hbs.md)
```
Exactly! Do you think that's good or bad? Do you reckon saying _"no"_ or ❌ out loud would help?

mamachanko commented on security-and-compliance/tls-and-certificates/ingress/inventory.hbs.md:
```diff
@@ -0,0 +1,64 @@
+
+To help with planning and accounting of TLS certificates, here's an inventory
+of TAP's components respective ingress:
+
```
At first I was sceptical about trimming away components with no ingress, what with it be _The Inventory_. Then it grew on me and I think you are making really good points! Thank you very much for digging into this.

kcoriordan commented on security-and-compliance/tls-and-certificates/ingress/inventory.hbs.md:
```diff
@@ -0,0 +1,64 @@
+carbonblack.scanning.apps.tanzu.vmware.com |  | | | | | | | |
+cartographer.tanzu.vmware.com | full, light, iterate, build, run | | | | | | | |
+cert-manager.tanzu.vmware.com | full, light, iterate, build, run, view | | | | | | | |
+cnrs.tanzu.vmware.com | full, light, iterate, run | ✅ | Instances of Knative's `Service` have ingress | ✅ | ✅ | `# of Services` | SANs depend on the component's `domain_template` (configurable) | [docs](../../../cloud-native-runtimes/about.hbs.md)
```
I wasn't sure tbh, so I left it as is.

kcoriordan commented on security-and-compliance/tls-and-certificates/ingress/inventory.hbs.md:
```diff
@@ -0,0 +1,64 @@
+services-toolkit.tanzu.vmware.com | full, light, iterate, run | | | | | | | |
+snyk.scanning.apps.tanzu.vmware.com | | | | | | | | |
+spring-boot-conventions.tanzu.vmware.com | full, light, iterate, build | | | | | | | |
+spring-cloud-gateway.tanzu.vmware.com | | ✅ | Instances of [SpringCloudGateway](../../../spring-cloud-gateway/about.hbs.md) have ingress | | ✅ | `# of SpringCloudGateway` | configurable | [docs](../../../spring-cloud-gateway/about.hbs.md) |
```
I added a link to this.

kcoriordan commented:
OK, made a few more updates not, I think it looks good, can you take another look to make sure you're happy with it
https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Application-Platform/1.6/tap/security-and-compliance-tls-and-certificates-ingress-inventory.html
@mamachanko

mamachanko commented:
> thanks Max @mamachanko - I can't change the background color unfortunately I made a few other changes based on a team edit review, to bring in line with style guide, accessibility, and translation standards:
> 
> -For the # of Ingress column – I changed that to 'number of ingress'. And had to make the items plural as a result. Can you check that's ok, and also, does this content need to be in code back ticks - I think they could be removed and it can be in normal text? I changed the variable values from <> to caps with dashes I changed the x and tick to Yes and No

@kcoriordan 👍🏻  to all your changes!

> does this content need to be in code back ticks - I think they could be removed and it can be in normal text?

💯 agreed!
